pkg/local/progress.go

```diff
@@ -133,14 +133,19 @@ func NewProgressPool() *ProgressPool {
 }
 
 type fileWrapper struct {
-	file   io.Seeker
-	reader io.Reader
+	file     io.Seeker
+	reader   io.Reader
+	readerAt io.ReaderAt
```

Comment:
There's no need for another field.
Define a new interface called `readerSeeker` that implements both io.Seeker and io.Reader and then use that type for the `file` parameter.